### PR TITLE
feature(views): the register form now has help text added to the fields

### LINKED ISSUE #4906: Include help text by registration form labels (Trac #4906)
_Original ticket http://trac.elgg.org/ticket/4906 on 42854588-05-31 by trac user mrclay, assigned to unknown._

Elgg version: 1.8

Should answer questions like:
* What is "Display name" mean?
* What kinds of chars can I use in the username?
* Password requirements?
* What happens when I click "Register"?

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #4906

register

If you have better texts... let me know

### CODE REVIEW

**Comment by hypeJunction** [comment]
Hm, not sure we need this in core. At least these should be easy to remove without having to overwrite the view. I guess you could always use css, but still how BC is this?

File: views/default/forms/register.php [hunk 1]
```diff
@@ -15,76 +15,77 @@
...
+
+$password2_input = elgg_view_input('password', [
+	'name' => 'password2',
+	'value' => '',
+	'label' => elgg_echo('passwordagain'),
+	'help' => elgg_echo('forms:register:password2:help'),
+	'required' => true,
+]);
+
+echo $name_input;
```
Comment (hypeJunction) [inline views/default/forms/register.php]: Why not echo right away?

Comment (jdalsem) [inline views/default/forms/register.php]: sure... no problem.

**Comment by jdalsem** [comment]
> I guess you could always use css

you can. I also thought about adding a view var to toggle the help 



> but still how BC is this

What BC issues are you thinking of..

**Comment by hypeJunction** [comment]
If I themed my registration form to fit in nicely in rectangle, this change would push the form out of the boundaries.

**Comment by jdalsem** [comment]
I don't think form heights and widths are part of the BC policy :smile:

File: languages/en.php [hunk 2]
```diff
@@ -397,6 +397,11 @@
 	'registerbad' => "Your registration was unsuccessful because of an unknown error.",
 	'registerdisabled' => "Registration has been disabled by the system administrator",
 	'register:fields' => 'All fields are required',
+	'forms:register:name:help' => 'This name is shown on your profile',
+	'forms:register:email:help' => 'Your username will be used for links to your profile',
+	'forms:register:username:help' => 'This email address will be used to send you mails. Other users can not see this email address',
```
Comment (hypeJunction) [inline languages/en.php]: email and username are reversed

**Comment by juho-jaakkola** [comment]
Let's not add features that are visible to end users in minor releases. All sites don't necessarily want to have them.

One option could be to add the logic with `elgg_language_key_exists('forms:register:foo:help')`, but not the actual translations. Sites that want the help texts could then simply add the translations in order to use the feature.

**Comment by mrclay** [comment]
> Let's not add features that are visible to end users in minor releases.

We need some wiggle room here. Probably a good discussion to have elsewhere.

**Comment by mrclay** [comment]
> One option could be to add the logic with elgg_language_key_exists('forms:register:foo:help'), but not the actual translations.

That sounds reasonable. We kinda need a simple framework to add "temporary" options for minor releases. Basically "These options will be on in 3.0, but can be enabled if you choose:"

**Comment by iionly** [comment]
Help texts for email address and username are swapped!

**Comment by jdalsem** [comment]
fixed swapped email and username

**Comment by mrclay** [comment]
If we don't think this is BC enough, another option is putting it in a bundled plugin called "3.0 Registration Form [experimental]". The description makes it clear that this may not be BC and is subject to change until 3.0.

**Comment by mrclay** [comment]
@jdalsem in #9713:
> we could toggle the visibility of the form fields with a view var that is default false. The plugin then could register a view var hook to turn it on.

Sounds like a good compromise. We should indicate that these options may go away or default on in 3.0, maybe even name it like `$vars['3.0 help_text']`.

**Comment by hypeJunction** [comment]
> we could toggle the visibility of the form fields with a view var that is default false. The plugin then could register a view var hook to turn it on.

Also think it's a good option. 
One reason I don't like this PR is the wording of help text. If I were a user I would think that the site owner is holding me for an idiot. I do know what passwords are needed for and what a username and email are. I think instead help text should indicate what validation rules apply:
`"The password must be at least 6 characters long and contain at least one digit ..."`
`"Must be a valid email address - it will be used to verify your account and let us communicate with you in case you forget your password"`

**Comment by hypeJunction** [comment]
I really don't want  this in core out of the box. If someone disagrees please reopen.

**Comment by jdalsem** [comment]
No problem.. i also closed related ticket

**Pull request closed without merging**